CMakeLists.txt

```diff
@@ -619,89 +619,189 @@ add_compile_definitions(
+# not enable terran or not rpc_found do
+# not ( enable terrain and rpc_found )
+# if ( NOT ( ${ENABLE_TERRAIN} AND ${RPC_FOUND} ) ) # this is wrong, needs fixing
+#   add_compile_definitions( LANDREAD_STUB )
+# endif()
+if ( ${ENABLE_TERRAIN} AND ${MOVE_NESTS} )
+  add_compile_definitions( TERRAIN_AND_LANDUSE )
+else ()
+  add_compile_definitions( LANDREAD_STUB )
+endif()
+
+if ( ${USE_ALLOCATABLES} )
+  add_compile_definitions( USE_ALLOCATABLES )
+endif()
+if ( ${wrfmodel} )
+  add_compile_definitions( wrfmodel )
+endif()
+if ( ${GRIB1} )
+  add_compile_definitions( GRIB1 )
+endif()
+if ( ${INTIO} )
+  add_compile_definitions( INTIO )
+endif()
+if ( ${KEEP_INT_AROUND} )
+  add_compile_definitions( KEEP_INT_AROUND )
+endif()
+if ( ${LIMIT_ARGS} )
+  add_compile_definitions( LIMIT_ARGS )
+endif()
+
+
+if ( ${BUILD_RRTMG_FAST} )
+  add_compile_definitions( BUILD_RRTMG_FAST=1 )
+else()
+  add_compile_definitions( BUILD_RRTMG_FAST=0 )
+endif()
+if ( ${BUILD_RRTMK} )
+  add_compile_definitions( BUILD_RRTMK=1 )
+else()
+  add_compile_definitions( BUILD_RRTMK=0 )
+endif()
+if ( ${BUILD_SBM_FAST} )
+  add_compile_definitions( BUILD_SBM_FAST=1 )
+else()
+  add_compile_definitions( BUILD_SBM_FAST=0 )
+endif()
+if ( ${SHOW_ALL_VARS_USED} )
+  add_compile_definitions( SHOW_ALL_VARS_USED=1 )
+else()
+  add_compile_definitions( SHOW_ALL_VARS_USED=0 )
+endif()
+if ( ${NMM_CORE} )
+  add_compile_definitions( NMM_CORE=1 )
+else()
+  add_compile_definitions( NMM_CORE=0 )
+endif()
+
+if ( "${WRF_CORE}" STREQUAL "PLUS" )
+  add_compile_definitions( WRFPLUS=1 )
+else()
+  add_compile_definitions( WRFPLUS=0 )
+endif()
+
+if ( "${WRF_CORE}" STREQUAL "DA_CORE" OR "${WRF_CORE}" STREQUAL "DA_4D_VAR" )
+  add_compile_definitions( DA_CORE=1 )
+else()
+  add_compile_definitions( DA_CORE=0 )
+endif()
+# DFI_RADAR=$<BOOL:${NMM_CORE}>
+
+# Nesting options
+if ( ${MOVE_NESTS} )
+  add_compile_definitions( MOVE_NESTS )
+endif()
+if ( "${WRF_NESTING}" STREQUAL "VORTEX" )
+  add_compile_definitions( VORTEX_CENTER )
+endif()
+
+# Configuration checks
+if ( NOT ${Fortran_2003_IEEE} )
+  add_compile_definitions( NO_IEEE_MODULE )
+endif()
+if ( NOT ${Fortran_2003_ISO_C} )
+  add_compile_definitions( NO_ISO_C_SUPPORT )
+endif()
+# If flush fails, check if we can fall back to fflush, and if not no support
+if ( NOT ${Fortran_2003_FLUSH} )
+  if ( "${Fortran_2003_FFLUSH}" )
+    add_compile_definitions( USE_FFLUSH )
+  else()
+    add_compile_definitions( NO_FLUSH_SUPPORT )
+  endif()
+endif()
+if ( NOT ${Fortran_2003_GAMMA} )
+  add_compile_definitions( NO_GAMMA_SUPPORT )
+endif()
+
+#!TODO Leaving as is in WRF for now but investigate why we don't do this
+# https://stackoverflow.com/a/1035713
+# If fseeko64 succeeds, use that, else check if we can fall back to fseeko, and if not just use fseek
+if ( NOT ${FSEEKO64} )
+  add_compile_definitions( FSEEKO64_OK )
+elseif( "${FSEEKO}" )
+  add_compile_definitions( FSEEKO_OK )
+else()
+  add_compile_definitions( FSEEK_OK )
+endif()
```

Comment:
Instead of calling add_compile_definitions for every "compile definition", I would append all compile definitions to a list and then call add_compile_definitions once using that list. At least for me, this would make the code easier to follow and it would be easy to print out all compile definitions via message.